tests/lib/rules/filename-naming-convention.posix.js

```diff
@@ -1271,3 +1271,95 @@ ruleTester.run(
     ],
   }
 );
+
+ruleTester.run(
+  "filename-naming-convention with option: [{ '**/components/featureA/*.*': '$1*' }]",
+  rule,
+  {
+    valid: [
+      {
+        code: "var foo = 'bar';",
+        filename: 'src/components/featureA/index.jsx',
+        options: [
+          { '**/components/**/*.*': '$1*' },
+          { ignoreMiddleExtensions: false, ignoreIndexFiles: true },
+        ],
+      },
+      {
+        code: "var foo = 'bar';",
+        filename: 'src/components/featureA/featureA.jsx',
+        options: [
+          { '**/components/**/*.*': '$1*' },
+          { ignoreMiddleExtensions: false },
+        ],
+      },
+      {
+        code: "var foo = 'bar';",
+        filename: 'src/components/featureA/featureA.specs.js',
+        options: [
+          { '**/components/**/*.*': '$1*' },
+          { ignoreMiddleExtensions: false },
+        ],
+      },
+      {
+        code: "var foo = 'bar';",
+        filename: 'src/components/featureA/featureA.specs.js',
+        options: [
+          { '**/components/**/*.*': '$1*' },
+          { ignoreMiddleExtensions: true },
+        ],
+      },
+    ],
```

Comment:
As shown in this test case and the one above, `ignoreMiddleExtensions` seems cant work well now.